libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -374,6 +375,13 @@ void IRGeneratorForStatements::endVisit(VariableDeclarationStatement const& _var
 {
 	setLocation(_varDeclStatement);
 
+	auto isTransient = [](auto const& _varDeclaration) { return (_varDeclaration ? _varDeclaration->referenceLocation() != VariableDeclaration::Location::Transient : true); };
```

Comment:
The name of the function does not match what it does :) For a moment I wanted to suggest it's buggy, but then realized you're checking the opposite.

Also, it seemed weird that you expect a value taken by reference to be nullable until I figured out the type. It's one of the cases where `auto` hides useful details and forces you to work harder to understand the code.
```suggestion
	auto static notTransient = [](std::shared_ptr<VariableDeclaration> const& _varDeclaration) {
		return !_varDeclaration || _varDeclaration->referenceLocation() != VariableDeclaration::Location::Transient;
	};
```

Reply:
> The name of the function does not match what it does :) For a moment I wanted to suggest it's buggy, but then realized you're checking the opposite.

Hahaha that's embarassing :sweat_smile: 

> Also, it seemed weird that you expect a value taken by reference to be nullable until I figured out the type. It's one of the cases where auto hides useful details and forces you to work harder to understand the code.

Ah yes, I should have realized that, because at first I didn't put the null check and only later inserted it because I got a seg fault when testing locally. The explicit type is much better.